### DISCUSSION TITLE
I2S to A2DP Source distorted audio

### DISCUSSION THREAD

**Discussion by nadnaks** [pschatzmann/ESP32-A2DP#196] [Q&A] [answered]
I'm trying to read an audio signal(a simple square wave tone from another ESP32) using I2S and send that signal to a Bluetooth speaker using A2DP source.  However, the audio is being distorted heavily.   The example we are using is base-I2S-A2DP from the audiotools library.   Do you have any ideas as to what could be causing the noise?

**Comment by pschatzmann** [comment, marked as answer]
I have never tried, but others seem to have issues as well: see https://github.com/pschatzmann/arduino-audio-tools/discussions/26
